x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -70,3 +178,66 @@ func createPoolRestriction(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 		return hasTwoCoins && hasPoolCreationFee
 	}
 }
+
+// getRandCLPool gets a concnerated liquidity pool with its pool denoms.
+func getRandCLPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (cltypes.ConcentratedPoolExtension, []string, error) {
+	// get all pools
+	clPools, err := k.GetAllPools(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	numPools := len(clPools)
+	if numPools == 0 {
+		return nil, nil, fmt.Errorf("no pools created")
+	}
+
+	pool_id := clPools[rand.Intn(numPools)].GetId()
+
+	// check if the pool exists
+	poolI, err := k.GetPool(ctx, pool_id)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Pool not found for id %d", pool_id)
+	}
+
+	concentratedPool, ok := poolI.(cltypes.ConcentratedPoolExtension)
+	if !ok {
+		return nil, nil, fmt.Errorf("interface conversion failed")
+	}
+
+	poolDenoms := []string{concentratedPool.GetToken0(), concentratedPool.GetToken1()}
+
+	return concentratedPool, poolDenoms, err
+}
+
+// getRandomTickPositions returns random lowerTick and upperTick divisible by tickSpacing value.
+func getRandomTickPositions(minTick, maxTick int64, tickSpacing uint64) (int64, int64, error) {
+	// Calculate the number of values between minTick and maxTick that are divisible by tickSpacing
+	lowerTickDivisible := ((maxTick - minTick) / int64(tickSpacing)) + 1
```

Comment:
Can you explain this to me a little further, for some reason it just simply isn't clicking. Max tick is 10 and min tick is 5 with a tick spacing of 1. This returns a lowerTickDivisible of 6. Why 6?

Reply:
I realized that this is a complex method. I found a simpler way :D 
we just keep generating random number from minTick to maxTick and take the first number divisible

Reply:
hopefully the new version is much clearer